packages/components/src/base-control/index.js

```diff
@@ -8,36 +8,47 @@ import classnames from 'classnames';
  */
 import VisuallyHidden from '../visually-hidden';
 
-function BaseControl( { id, label, hideLabelFromVision, help, className, children } ) {
+function BaseControl( { as = 'div', id, label, hideLabelFromVision, help, className, children } ) {
+	const isFieldSet = as === 'fieldset';
+
+	/**
+	 * BaseControl can only render as either a fieldset or a div
+	 */
+	const WrapperComponent = as;
+	const LabelComponent = isFieldSet ? 'legend' : 'label';
+	const VisualLabelComponent = isFieldSet ? 'legend' : 'span';
+
 	return (
-		<div className={ classnames( 'components-base-control', className ) }>
+		<WrapperComponent className={ classnames( 'components-base-control', className ) }>
 			<div className="components-base-control__field">
 				{ label && id && ( hideLabelFromVision ?
 					<VisuallyHidden
 						as="label"
 						htmlFor={ id }>{ label }</VisuallyHidden> :
-					<label
+					<LabelComponent
 						className="components-base-control__label"
-						htmlFor={ id }>{ label }</label>
+						htmlFor={ id }>{ label }</LabelComponent>
```

Comment:
The "for" attribute should only be applied to a label element, not a legend element. Especially if that legend element's for IDRef only references the fieldset it is contained within. See the storybook tests at the end of the patch for what I mean. Other than that, the patch looks awesome.

Reply:
@MarcoZehe Oo! Thank you for catching that. I'll make the adjustment for the `for` attribute.